code/game/machinery/constructable_frame.dm

```diff
@@ -402,6 +405,18 @@
 	if(user.combat_mode)
 		return ..()
 
+/// returns instance of path1 in list else path2 in list
+/obj/structure/frame/machine/proc/look_for(list/parts, path1, path2 = null)
+	//look for path1 in list
+	var/part = locate(path1) in parts
+	if(!isnull(part))
+		return part
+
+	//optional look for path2 in list
+	if(!isnull(path2))
+		part = locate(path2) in parts
+	return part
```

Comment:
This could be using args and ... instead.
And I would also recommend a much more indepth codedoc

Reply:
This is just a general purpose helper proc and isnt tied to the machinery code so don't know what more I could add to the code doc. 
Using args for just 2 params seems a bit excessive